daliuge-engine/dlg/data/drops/file.py

```diff
@@ -70,24 +78,57 @@ def __init__(self, *args, **kwargs):
             kwargs["persist"] = True
         if kwargs["persist"] and "lifespan" not in kwargs:
             kwargs["expireAfterUse"] = False
+        self.is_dir = False
         super().__init__(*args, **kwargs)
 
-    def sanitize_paths(self, filepath):
+    def sanitize_paths(self, filepath: str) -> Union[None, str]:
+        """
+        Expand ENV_VARS, but also deal with relative
+        and absolute paths. filepath can be either just be
+        a directory, a directory including a file name, only
+        a directory (both relative and absolute), or just
+        a file name.
+
+        :param filepath: string, path and or directory
 
+        :returns filepath
+        """
         # replace any ENV_VARS on the names
-        return os.path.expandvars(filepath) if filepath else None
+        if not filepath:
+            return None
+        filepath = os.path.expandvars(filepath)
+        if isabs(filepath):
+            return filepath
+        else:
+            filepath = self.get_dir(filepath)
+        return filepath
 
     non_fname_chars = re.compile(r":|%s" % os.sep)
 
     def initialize(self, **kwargs):
         """
         FileDROP-specific initialization.
         """
-        self.filepath = self.parameters.get("filepath", None)
-
-        filepath = self.sanitize_paths(self.filepath)
-        filename = os.path.basename(filepath) if filepath else None
-        dirname = os.path.dirname(filepath) if filepath else None
+        filepath = self.parameters.get("filepath", None)
+        dirname = None
+        filename = None
+
+        if filepath:  # if there is anything provided
+            if filepath.count("/") == 0:  # just a name
```

Comment:
Purely stylistic, but it reads slightly better to the casual reader:

```suggestion
            if "/" not in filepath:  # just a name
```